dart-lang/webdev
# Add Webdev installation test

Test has 3 steps:
- **Activate webdev**: if activation times out after 30 seconds, fails with `stdout` and `stderr` as reason (for debugging)
- **Create a new web app**: if creation times out after 30 seconds, fails with `stdout` and `stderr` as reason (for debugging)
-  **Serve the app with `webdev serve`**:  kills the process after the stdout includes `serving {{app}} on` or after 30 seconds. If killed after 30 seconds, we fail with the stdout and stderr  as reason (for debugging)

See for example (failure from timeout):


```
  Expected: contains 'Serving `web` on'
    Actual: '\x1B[2K\r[INFO] Connecting to the build daemon...\x1B[2K\r[INFO] Generating build script...\x1B[2K\r[INFO] Generating build script completed, took 363ms\x1B[2K\r[INFO] \x1B[2K\r[INFO] Precompiling build script......'
     Which: does not contain 'Serving `web` on'
  stderr:

  package:matcher                    expect
  test/installation_test.dart 67:5   main._expectStdoutThenExit
  ===== asynchronous gap ===========================
  test/installation_test.dart 130:5  main.<fn>

```
